components/org.wso2.carbon.das.jobmanager.core/src/main/java/org/wso2/carbon/das/jobmanager/core/topology/SiddhiTopologyCreatorImpl.java

```diff
@@ -364,124 +359,178 @@ private void checkQueryType(InputStream inputStream, boolean queryElement) {
+                                    siddhiQueryGroup2.getInputStreams().get(key).getSubscriptionStrategy();
                             if (subscriptionStrategy.getStrategy().equals(TransportStrategy.FIELD_GROUPING)) {
-
                                 fieldGrouping = true;
-                                for (PublishingStrategyDataHolder publishingStrategyDataHolder : siddhiQueryGroupsList
-                                        .get(i).getOutputStream().get(key).getPublishingStrategyList()) {
+                                for (PublishingStrategyDataHolder publishingStrategyDataHolder :
+                                        siddhiQueryGroup1.getOutputStream().get(key).getPublishingStrategyList()) {
 
                                     if (publishingStrategyDataHolder.getGroupingField() != null
                                             && publishingStrategyDataHolder.getGroupingField()
-                                            .equals(siddhiTopologyDataHolder.getPartitionTypeMap().get(key)
+                                            .equals(siddhiTopologyDataHolder.getPartitionKeyMap().get(key)
                                                             .getFirst())) {
 
-                                        int index = siddhiQueryGroupsList.get(i).getOutputStream().get(key)
+                                        int index = siddhiQueryGroup1.getOutputStream().get(key)
                                                 .getPublishingStrategyList().indexOf(publishingStrategyDataHolder);
 
                                         //check partition-key if same ---> need to check for max-No of groups
-                                        siddhiQueryGroupsList.get(i).getOutputStream().get(key)
+                                        siddhiQueryGroup1.getOutputStream().get(key)
                                                 .getPublishingStrategyList().get(index).setParallelism(
-                                                Math.max(siddhiQueryGroupsList.get(i).getParallelism(),
-                                                         siddhiQueryGroupsList.get(j).getParallelism()));
+                                                Math.max(siddhiQueryGroup1.getParallelism(),
+                                                         siddhiQueryGroup2.getParallelism()));
 
-                                        siddhiQueryGroupsList.get(j).getInputStreams().get(key)
+                                        siddhiQueryGroup2.getInputStreams().get(key)
                                                 .getSubscriptionStrategy().setOfferedParallelism(
-                                                Math.max(siddhiQueryGroupsList.get(i).getParallelism(),
-                                                         siddhiQueryGroupsList.get(j).getParallelism()));
+                                                Math.max(siddhiQueryGroup1.getParallelism(),
```

Comment:
Are we handling the scenario of having n partition subscribers with different parallelism.

Reply:
yes..n partition subscribers with different parallelism for the same key